src/OpenTelemetry/Metrics/MetricPointOptionalComponents.cs

```diff
@@ -38,4 +40,30 @@ internal MetricPointOptionalComponents Copy()
 
         return copy;
     }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public void AcquireLock()
+    {
+        if (Interlocked.Exchange(ref this.isCriticalSectionOccupied, 1) != 0)
+        {
+            this.AcquireLockRare();
```

Comment:
Would this really be rare? 😄

Reply:
Define rare 😄

Let me explain why I split this method up. Before we had this:

```csharp
        var sw = default(SpinWait);
        while (Interlocked.Exchange(ref isCriticalSectionOccupied, 1) != 0)
        {
            sw.SpinOnce();
        }
```

C# initializes all locals by default. Part of its safety system. So we pay for the initialization of `sw` even thought we might not use it. What the split does is make it optimistic essentially. If we can get the lock, we're done. Otherwise we jump into a more rare case where we pay for the `SpinWait` because we know we need it.

For the thread that got the lock, it saved a tiny bit of CPU so it can release the lock even faster. For threads that need to wait, we're going to spin them anyway so it seems OK to me to ask them to do an extra call/jump. Thoughts?

Reply:
> Define rare 😄

I think having apps where multiple threads update the same MetricPoint would be a common thing.

> For the thread that got the lock, it saved a tiny bit of CPU so it can release the lock even faster.

I don't think splitting the acquire lock code would speed up the release. In the existing code, the thread that acquires the lock would have already created the `SpinWait` struct before acquiring the lock. So, nothing has changed in what happens after a thread takes the lock. Or would the speed up come from using `if` vs `while`?
 
Having said all that, I'm okay with this change to avoid creating `SpinWait` struct for the threads that need to wait.